### ISSUE TITLE
Reminders no longer trigger after x minutes

### ISSUE THREAD

**Issue by onionhammer** [dotnet/orleans#4432]
After upgrading from Orleans RC2 to Orleans 2.0 final, table storage reminders have been working intermittently, but more often they do not work

**Comment by onionhammer** [comment]
## Initialization

```C#
Silo = new SiloHostBuilder()
  ...
  .UseAzureStorageClustering(options => options.ConnectionString = Options.DataConnectionString)
  .UseAzureTableReminderService(Options.DataConnectionString)
  ...
  .Build();
```

## Usage
From our `StatusTrackingGrain` we set a reminder to trigger after 10 minutes

```C#
await RegisterOrUpdateReminder(SendEvent, span, span);
```

After that in the `OrleansReminders` table I see
![image](https://user-images.githubusercontent.com/969938/38509842-b519b572-3be8-11e8-8629-63950d799d13.png)

```C#
        public async Task ReceiveReminder(string reminderName, TickStatus status)
        {
              // Clear reminder registration
              // Log event
        } 
```

Note it's the same exact code that works in RC2 and does not work in 2.0

As far as I can see, no errors are getting logged

**Comment by onionhammer** [comment]
Maybe it's related to this? https://github.com/dotnet/orleans/commit/976ac32887ebcab0fd685d61873f4897f4217a8a#diff-4632ee844ec36021b8fb4ed9874047e0

**Comment by onionhammer** [comment]
Okay, after adding a `ServiceId` it now works. The application seems to work perfectly fine *other* than reminders without specifying a `ServiceId`

```C#
 Silo = new SiloHostBuilder()
     .Configure((ClusterOptions options) =>
     {
         options.ClusterId = SiloLocator.GetClusterId();
         options.ServiceId = "MyApp";
     })
```

**Comment by sergeybykov** [comment]
Looks like a duplicate of #4401 to me.

**Comment by sergeybykov** [comment]
>@sergeybykov it could be just another symptom of that. Perhaps ServiceId should default to something?

We used to implicitly default it to empty GUID. I think the right solution is to require it. We are just missing validation in the builder.

**Comment by onionhammer** [comment]
@sergeybykov hmm but it's a string not a guid. Why would empty guid not work for reminders but specifying a hardcoded string work?

**Comment by sergeybykov** [comment]
Yes, we changed it in 2.0 from GUID to string, because we it was misleading people to think that Service ID is something that you generate for each cluster deployment, as opposed effectively an Application ID that needs to stay the same across the lifecycle of the service/app, upgrades and all.

The fact that `Guid` is a value type only added to the confusion with its default value of empty GUID. 

That's why we believe Service ID  should be explicitly set and validated by the builder.
